pkg/gateway/model_build_targetgroup.go

```diff
@@ -350,22 +358,27 @@ func (t *latticeServiceModelBuildTask) buildTargetGroupSpec(ctx context.Context,
 		}
 	}
 
-	tgName := latticestore.TargetGroupName(string(httpBackendRef.Name()), namespace)
+	tgName := latticestore.TargetGroupName(string(backendRef.Name()), namespace)
+
+	protocolVersion := vpclattice.TargetGroupProtocolVersionHttp1
+	if _, ok := t.route.(*core.GRPCRoute); ok {
+		protocolVersion = vpclattice.TargetGroupProtocolVersionGrpc
+	}
```

Comment:
probably switch on route type with default to http1

Reply:
Is that not the same thing?